tests/integrationv2/test_happy_path.py

```diff
@@ -16,16 +16,22 @@ def test_s2n_server_happy_path(managed_process, cipher, curve, provider):
     host = "localhost"
     port = next(available_ports)
 
+    # S2ND can receive large amounts of data because all the data is
+    # echo'd to stdout unmodified. This lets us compare received to
+    # expected easily.
+    random_bytes = data_bytes(16384)
```

Comment:
This isn't a lot. Do you want to have more to actually see whether multiple fragments are handled correctly? If there is no general issue with sending more bytes, sending a few megabytes should still be fast and give a lot better coverage.

Reply:
We do check to make sure the data was received. We do a simple comparison though, just looking at whether what the client sent was received by the server.

You have a good point about the amount of data. It would be unreasonable to use our current check to do this. Unfortunately, we would need a different way for the client and server to store the data they are receiving. I've created an issue for this, #1844. It would be nice to add some network congestion during these tests as well.

Reply:
I bumped the data send to 4mB here. I was worried about a failure causing all that data to be dumped, making the root cause very hard to find. But the happy path test is probably not where we are going to fail. We can use smaller sizes in other tests.